cmd/manager/main.go

```diff
@@ -98,6 +99,13 @@ func main() {
 	go registrySyncer.Sync(stopCh)
 	go catalogsourceconfig.Syncer.Sync(stopCh)
 
+	if os.Getenv("RELEASE_VERSION") != "" {
+		client, err := client.New(config.GetConfigOrDie(), client.Options{})
+		clean := NewCleanUpResources(log.NewEntry(log.New()), client)
```

Comment:
You need to update the `ClusterOperator` CR during this process to show that we are performing some upgrade logic.

Reply:
We also want to make sure that the `ClusterOperator` is not set to available while resources are being migrated - If the Upgrade path is 4.1->4.2->4.3, we don't want the upgrade from 4.2->4.3 to kickoff before this completes.

Reply:
Each cluster upgrade is a distinct operation. I don't think the 4.2 --> 4.3 option will be available until 4.1 --> 4.2 is complete.

Reply:
I agree - but if marketplace reports that progression has finished we risk a race condition.

Reply:
I don't understand how there is a race condition. If marketplace has not finished migration, then the cluster is not ready for the next upgrade.